pkg/decoders/escaped_unicode.go

```diff
@@ -0,0 +1,65 @@
+package decoders
+
+import (
+	"regexp"
+	"strconv"
+	"unicode/utf8"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/sources"
+)
+
+type EscapedUnicode struct{}
+
+var _ Decoder = (*EscapedUnicode)(nil)
+
+// It might be advantageous to limit these to a subset of acceptable characters, similar to base64.
+// https://dencode.com/en/string/unicode-escape
+var escapePat = regexp.MustCompile(`(?i:\\{1,2}u)([a-fA-F0-9]{4})`)
+
+func (d *EscapedUnicode) FromChunk(chunk *sources.Chunk) *DecodableChunk {
+	if chunk == nil || len(chunk.Data) == 0 || !escapePat.Match(chunk.Data) {
+		return nil
+	}
+
+	chunk.Data = decodeUnicode(chunk.Data)
+	decodableChunk := &DecodableChunk{
+		DecoderType: detectorspb.DecoderType_ESCAPED_UNICODE,
+		Chunk:       chunk,
+	}
+
+	return decodableChunk
+}
+
+func decodeUnicode(input []byte) []byte {
+	// Find all Unicode escape sequences in the input byte slice
+	indices := escapePat.FindAllSubmatchIndex(input, -1)
+
+	// Iterate over found indices in reverse order to avoid modifying the slice length
+	//for i, matches := range indices {
+	for i := len(indices) - 1; i >= 0; i-- {
+		matches := indices[i]
+		startIndex := matches[0]
+		hexStartIndex := matches[2]
+		endIndex := matches[3]
+
+		// Extract the hexadecimal value from the escape sequence
+		hexValue := string(input[hexStartIndex:endIndex])
+
+		// Parse the hexadecimal value to an integer
+		unicodeInt, err := strconv.ParseInt(hexValue, 16, 32)
+		if err != nil {
+			// If there's an error, continue to the next escape sequence
+			continue
+		}
+
+		// Convert the Unicode code point to a UTF-8 representation
+		utf8Bytes := make([]byte, 4)
```

Comment:
question: Currently, the allocation for utf8Bytes occurs within each iteration of the loop. Would it be possible to allocate this slice a single time before the loop starts? This change would mean we only incur the allocation cost once, potentially enhancing the function's efficiency, especially for inputs with numerous Unicode escape sequences.

super nit: Regarding the 4 used in the slice allocation, it might not be immediately clear to all readers that this represents the maximum number of bytes required to encode a single Unicode rune in UTF-8. To make this more explicit and enhance code readability, I suggest defining a constant at the beginning of the function. For example:
`const maxBytesPerRune = 4 // Maximum number of bytes to encode a single Unicode rune in UTF-8`

Reply:
> question: Currently, the allocation for utf8Bytes occurs within each iteration of the loop. Would it be possible to allocate this slice a single time before the loop starts? This change would mean we only incur the allocation cost once, potentially enhancing the function's efficiency, especially for inputs with numerous Unicode escape sequences.

That should be fine. While [`utf8.EncodeRune`](https://github.com/golang/go/blob/7db8b9e518311c1ee707a3f2fe6343e7c595c936/src/runtime/utf8.go#L104) doesn't clear the slice, it returns the length of bytes written.